### ISSUE TITLE
Decouple communication between presentation layer (CLI/Service) and renku core

### ISSUE THREAD

**Issue by Panaetius** [SwissDataScienceCenter/renku-python#737]
In #702 dataset import/export click dependencies were removed from `renku.core` code, by using callbacks and otherwise injecting the relevant code into the business layer.

This works, but can get a bit unwieldy, especially since many of the callbacks are still somewhat `click`-specific and the interface might not be applicable for other potential consumers of `core`. Additionally, each method has it's own callbacks that sometimes differ from other functions' callbacks, making them still conceptually tightly coupled.

Basically, the refactoring we did with the `cli`, `core` namespaces means our application now has a rather traditional `presentation`, `logic` and potentially `DAL` layered architecture, so we should follow patterns that make sense to decouple `presentation` from `logic`.

Something like the mediator pattern or event based communication could work well.

As this should be unified for all functions (not each having its own slightly different implementation), it needs some architecture work and is not done in #702

**Comment by Panaetius** [comment]
This has been implemented
